### ISSUE TITLE
Win64 py2 error with git source

### ISSUE THREAD

**Issue by rmcgibbo** [conda/conda-build#633]
I just saw the following error on Appveyor when building [this recipe](https://github.com/rmcgibbo/conda-recipes/blob/b568327382194d1b03bebdf437131b7032ae5f14/cclib/meta.yaml) with conda-build 1.18
```
BUILD START: cclib-1.3.1-py27_0
Fetching package metadata: .......... 
Solving package specifications: ..............Cloning into bare repository '/cygdrive/c/Miniconda-x64/conda-bld/git_cache/__github.com_cclib_cclib.git'...
Cloning into 'C:\Miniconda-x64\conda-bld\work'...
done.
Note: checking out 'v1.3.1'.
 
You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.
 
If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:
 
  git checkout -b new_branch_name
 
HEAD is now at 58f3dda... New DOI from Zenodo
 
The following NEW packages will be INSTALLED:
 
    pip:        7.1.2-py27_0 
    python:     2.7.10-1     
    setuptools: 18.3.2-py27_0
    wheel:      0.26.0-py27_1
 
Removing old work directory
checkout: u'v1.3.1'
An unexpected error has occurred, please consider sending the
following traceback to the conda GitHub issue tracker at:
 
    https://github.com/conda/conda-build/issues
 
Include the output of the command 'conda info' in your report.
 
 
Traceback (most recent call last):
  File "C:\Miniconda-x64\Scripts\conda-build-script.py", line 4, in <module>
    sys.exit(main())
  File "C:\Miniconda-x64\lib\site-packages\conda_build\main_build.py", line 190, in main
    args_func(args, p)
  File "C:\Miniconda-x64\lib\site-packages\conda_build\main_build.py", line 468, in args_func
    args.func(args, p)
  File "C:\Miniconda-x64\lib\site-packages\conda_build\main_build.py", line 391, in execute
    override_channels=args.override_channels, include_recipe=args.include_recipe)
  File "C:\Miniconda-x64\lib\site-packages\conda_build\build.py", line 392, in build
    source.provide(m.path, m.get_section('source'))
  File "C:\Miniconda-x64\lib\site-packages\conda_build\source.py", line 269, in provide
    git_source(meta, recipe_dir)
  File "C:\Miniconda-x64\lib\site-packages\conda_build\source.py", line 136, in git_source
    git_info()
  File "C:\Miniconda-x64\lib\site-packages\conda_build\source.py", line 160, in git_info
    stdout = stdout.decode(encoding, 'ignore')
TypeError: decode() argument 1 must be string, not None
```

**Comment by rmcgibbo** [comment]
It's very strange that encoding could be `None` here, given the context: https://github.com/conda/conda-build/blob/master/conda_build/source.py#L158-L160

**Comment by rmcgibbo** [comment]
Fixed by #636
